Frameworks/CoreData/NSEntityDescription.mm

```diff
@@ -14,10 +14,164 @@
 //
 //******************************************************************************
 
+#import <CoreData/CoreData.h>
+#import <Foundation/Foundation.h>
 #import <StubReturn.h>
-#import <CoreData/NSEntityDescription.h>
+
+#import <CoreData/NSManagedObjectModel-XMLParsing.h>
+
+@interface NSEntityDescription () {
+    NSManagedObjectModel* _managedObjectModel;
+    StrongId<NSArray<NSPropertyDescription*>> _properties;
+    StrongId<NSArray<NSEntityDescription*>> _subentities;
+    StrongId<NSMutableArray<NSPropertyDescription*>> _unresolvedProperties;
+}
+@property (readwrite, assign) NSEntityDescription* superentity;
+@property (readwrite, copy) NSDictionary* subentitiesByName;
+@property (readwrite, copy) NSDictionary* propertiesByName;
+@property (readwrite, copy) NSDictionary* attributesByName;
+@property (readwrite, copy) NSDictionary* relationshipsByName;
+@end
 
 @implementation NSEntityDescription
+- (id)initWithXMLElementName:(NSString*)entityName attributes:(NSDictionary<NSString*, NSString*>*)attributes {
+    if (self = [super init]) {
+        _unresolvedProperties = [NSMutableArray new];
+        _name = [attributes[@"name"] copy];
+        _managedObjectClassName = [attributes[@"representedClassName"] copy];
+        _renamingIdentifier = [attributes[@"elementID"] copy]; // TODO(DH): figure.
+        _versionHashModifier = [attributes[@"versionHashModifier"] copy];
+        _abstract = [attributes[@"isAbstract"] isEqualToString:@"YES"];
+
+        // _userInfo, _compoundIndexes, and _uniquenessConstraints are set out-of-band by the XML parser.
+        // _subentities, *ByName, and other collection properties are derived from sub-elements via the XML parser.
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [_name release];
+    [_managedObjectClassName release];
+    [_renamingIdentifier release];
+    [_userInfo release];
+    [_compoundIndexes release];
+    [_uniquenessConstraints release];
+    [_versionHash release];
+    [_versionHashModifier release];
+
+    // derived properties
+    [_subentitiesByName release];
+    [_propertiesByName release];
+    [_attributesByName release];
+    [_relationshipsByName release];
```

Comment:
This makes me so sad. I suppose this is due to synthesized properties but :-( ... RAII would be much nicer.